runtime/src/impls.rs

```diff
@@ -81,14 +81,22 @@ impl Convert<u128, Balance> for CurrencyToVoteHandler {
 
 /// Convert from weight to balance via a simple coefficient multiplication
 /// The associated type C encapsulates a constant in units of balance per weight
-pub struct LinearWeightToFee<C>(sp_std::marker::PhantomData<C>);
+pub struct ScaleLinearWeightToFee<C>(sp_std::marker::PhantomData<C>);
 
-impl<C: Get<Balance>> Convert<Weight, Balance> for LinearWeightToFee<C> {
+impl<C: Get<Balance>> Convert<Weight, Balance> for ScaleLinearWeightToFee<C> {
 	fn convert(w: Weight) -> Balance {
 		// cennznet-node a weight of 10_000 (smallest non-zero weight) to be mapped to 10^7 units of
 		// fees, hence:
 		let coefficient = C::get();
-		Balance::from(w).saturating_mul(coefficient)
+		let linear_weight = Balance::from(w).saturating_mul(coefficient);
+		// Scale down fees 10^-12
+		let fee_divider: Balance = 1_000_000_000_000;
```

Comment:
I have added a parameter called ScaleDownFactor to Runtime for this purpose. If I merge first, you can reuse that.

Reply:
done

Reply:
Now after "**use**"ing the ScaleDownFactor here, you can do `let fee_divider = ScaleDownFactor::get();`